v2/protocol/http/protocol_lifecycle.go

```diff
@@ -38,10 +38,8 @@ func (p *Protocol) OpenInbound(ctx context.Context) error {
 	}
 
 	p.server = &http.Server{
-		Addr:         listener.Addr().String(),
-		Handler:      attachMiddleware(p.Handler, p.middleware),
-		ReadTimeout:  DefaultTimeout,
-		WriteTimeout: DefaultTimeout,
+		Addr:    listener.Addr().String(),
+		Handler: attachMiddleware(p.Handler, p.middleware),
```

Comment:
question: why are you not setting the timeouts here?

Reply:
@embano1 similar to `ShutdownTimeout` the default is set here:

```
if p.WriteTimeout == 0 {
		p.WriteTimeout = DefaultTimeout
	}
```

From looking at the code it seemed apparent to me that by setting the timeout in the OpenInboundConnection would override anything set by the `Protocol`. Which would effectively render my change useless.

Also, how do you want to distinguish between the user explicitly setting a `0` or `negative` value, vs, they just want the default to be set? Should be make 0 -> DefaultTimeout, and a negative value will just be passed through to indicate no timeout?